### PR TITLE
feat(pipeline): edge pipeline store access key in etcd

### PR DESCRIPTION
**Pull request by chengjoey**
#### What this PR does / why we need it:
edge pipeline store access key in etcd

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English | edge pipeline store access key in etcd             |
| 🇨🇳 中文    |   边缘流水线存储access key进行校验           |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: modules/pipeline/providers/edgepipeline_register/interface.go [hunk 1]
```diff
@@ -134,8 +134,11 @@ func (p *provider) CheckAccessTokenFromHttpRequest(req *http.Request) error {
 }
 
 func (p *provider) CheckAccessToken(token string) error {
-	if token != p.EdgeTaskAccessToken() {
-		return fmt.Errorf("invalid access token")
+	accessEtcdKey := p.makeClusterAccessEtcdKey(token)
```
Comment (sfwn) [inline modules/pipeline/providers/edgepipeline_register/interface.go:137]: makeEtcdKeyOfClusterAccessKey

File: modules/pipeline/providers/edgepipeline_register/interface.go [hunk 2]
```diff
@@ -111,15 +111,15 @@ func (p *provider) GetEdgeBundleByClusterName(clusterName string) (*bundle.Bundl
 
 func (p *provider) GetAccessToken(req apistructs.OAuth2TokenGetRequest) (*apistructs.OAuth2Token, error) {
 	return &apistructs.OAuth2Token{
-		AccessToken: p.EdgeTaskAccessToken(),
+		AccessToken: p.ClusterAccessKey(),
```
Comment (sfwn) [inline modules/pipeline/providers/edgepipeline_register/interface.go:114]: split EdgeTaskAccessToken to a new file.
or delete it if useless.

File: modules/pipeline/providers/edgepipeline_register/interface.go [hunk 3]
```diff
@@ -111,15 +111,15 @@ func (p *provider) GetEdgeBundleByClusterName(clusterName string) (*bundle.Bundl
...
 		ExpiresIn:   0,
 		TokenType:   "Bearer",
 	}, nil
 }
 
 func (p *provider) GetOAuth2Token(req apistructs.OAuth2TokenGetRequest) (*apistructs.OAuth2Token, error) {
 	return &apistructs.OAuth2Token{
-		AccessToken: p.EdgeTaskAccessToken(),
+		AccessToken: p.ClusterAccessKey(),
```
Comment (sfwn) [inline modules/pipeline/providers/edgepipeline_register/interface.go:122]: split all cluster access key related to a new file.

File: modules/pipeline/providers/edgepipeline_register/provider.go [hunk 4]
```diff
@@ -36,7 +38,7 @@ type Config struct {
 	ClusterDialEndpoint        string        `file:"cluster_dialer_endpoint" desc:"cluster dialer endpoint"`
 	ClusterAccessKey           string        `file:"cluster_access_key" desc:"cluster access key, if specified will doesn't start watcher"`
 	RetryConnectDialerInterval time.Duration `file:"retry_cluster_hook_interval" default:"1s"`
-	AccessToken                string        `file:"token" env:"EDGE_PIPELINE_ACCESS_TOKEN"`
+	AccessClusterEtcdKeyPrefix string        `env:"EDGE_PIPELINE_ACCESS_CLUSTER_ETCD_KEY_PREFIX" default:"/devops/pipeline/v2/edge/pipeline/access-key"`
```
Comment (sfwn) [inline modules/pipeline/providers/edgepipeline_register/provider.go:41]: you have to add file config.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/pipeline/providers/edgepipeline_register/interface.go (hunk 1 above)
Comment (chengjoey) [reply modules/pipeline/providers/edgepipeline_register/interface.go]: adjusted

File: modules/pipeline/providers/edgepipeline_register/interface.go (hunk 2 above)
Comment (chengjoey) [reply modules/pipeline/providers/edgepipeline_register/interface.go]: splited

File: modules/pipeline/providers/edgepipeline_register/interface.go (hunk 3 above)
Comment (chengjoey) [reply modules/pipeline/providers/edgepipeline_register/interface.go]: splited

File: modules/pipeline/providers/edgepipeline_register/provider.go (hunk 4 above)
Comment (chengjoey) [reply modules/pipeline/providers/edgepipeline_register/provider.go]: added

**Review by sfwn — APPROVED** [review]

File: conf/pipeline/pipeline.yaml [hunk 5]
```diff
@@ -41,6 +41,7 @@ edgepipeline_register:
   cluster_dialer_endpoint: "${CLUSTER_DIALER_PUBLIC_URL}/clusteragent/connect"
   cluster_access_key: "${CLUSTER_ACCESS_KEY}"
   allowed_sources: "cdp-,fdp-"
+  edge_pipeline_access_cluster_etcd_key_prefix: "${EDGE_PIPELINE_ACCESS_CLUSTER_ETCD_KEY_PREFIX:/devops/pipeline/v2/edge/pipeline/access-key}"
```
Comment (sfwn) [inline conf/pipeline/pipeline.yaml:44]: already in a edgepipeline_register scope.
so key should be: cluster_access_key_etcd_prefix
env should be: EDGE_PIPELINE_CLUSTER_ACCESS_KEY_ETCD_PREFIX
default: devops/pipeline/v2/edge/cluster-access-key

File: modules/pipeline/providers/edgepipeline_register/provider.go [hunk 6]
```diff
@@ -36,7 +38,7 @@ type Config struct {
 	ClusterDialEndpoint        string        `file:"cluster_dialer_endpoint" desc:"cluster dialer endpoint"`
 	ClusterAccessKey           string        `file:"cluster_access_key" desc:"cluster access key, if specified will doesn't start watcher"`
 	RetryConnectDialerInterval time.Duration `file:"retry_cluster_hook_interval" default:"1s"`
-	AccessToken                string        `file:"token" env:"EDGE_PIPELINE_ACCESS_TOKEN"`
+	AccessClusterEtcdKeyPrefix string        `file:"edge_pipeline_access_cluster_etcd_key_prefix" env:"EDGE_PIPELINE_ACCESS_CLUSTER_ETCD_KEY_PREFIX" default:"/devops/pipeline/v2/edge/pipeline/access-key"`
```
Comment (sfwn) [inline modules/pipeline/providers/edgepipeline_register/provider.go:41]: no need default value here.
only define once in config.yaml or provider.config.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: conf/pipeline/pipeline.yaml (hunk 5 above)
Comment (chengjoey) [reply conf/pipeline/pipeline.yaml]: adjusted

File: modules/pipeline/providers/edgepipeline_register/provider.go [hunk 7]
```diff
@@ -36,7 +38,7 @@ type Config struct {
 	ClusterDialEndpoint        string        `file:"cluster_dialer_endpoint" desc:"cluster dialer endpoint"`
 	ClusterAccessKey           string        `file:"cluster_access_key" desc:"cluster access key, if specified will doesn't start watcher"`
 	RetryConnectDialerInterval time.Duration `file:"retry_cluster_hook_interval" default:"1s"`
-	AccessToken                string        `file:"token" env:"EDGE_PIPELINE_ACCESS_TOKEN"`
+	AccessClusterEtcdKeyPrefix string        `file:"cluster_access_key_etcd_prefix" env:"EDGE_PIPELINE_CLUSTER_ACCESS_KEY_ETCD_PREFIX"`
```
Comment (sfwn) [inline modules/pipeline/providers/edgepipeline_register/provider.go:41]: you should check prefix should not be empty when provider init.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/pipeline/providers/edgepipeline_register/provider.go (hunk 7 above)
Comment (sfwn) [reply modules/pipeline/providers/edgepipeline_register/provider.go:41]: also check ending slash

Comment (chengjoey) [reply modules/pipeline/providers/edgepipeline_register/provider.go]: checked

**Review by sfwn — APPROVED** [review]

**Pull request merged**